### PR TITLE
post: "how we built this" w/Brave

### PR DESCRIPTION
**Pull request by jennwrites**
Second post of the Brave campaign to be published at 1pm Eastern time today.

- [x] Confirm author
- [x] Confirm whether we need screenshots/gifs of integration in action

### CODE REVIEW

**Comment by jennwrites** [comment]
Hey @mitchwagner can you submit a patch here?

**Review by autonome — APPROVED** [review]
Looks good, create to see the historical context. Please update the title to a single action statement.

File: content/post/119-how-we-built-this.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+date: 2021-01-21
+url: /2021-01-21-how-we-built-this/
+title: Integrating IPFS and Brave – How We Built This
```
Comment (lidel) [inline content/post/119-how-we-built-this.md]: This is just a proposal – feel free to tweak.
It was team effort, no need to be so serious :)

```suggestion
title: How We Put IPFS in Brave
```

File: content/post/119-how-we-built-this.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+date: 2021-01-21
+url: /2021-01-21-how-we-built-this/
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:3]:
```suggestion
url: /2021-01-21-how-we-put-ipfs-in-brave/
```

File: content/post/119-how-we-built-this.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Collaboration
+Both Brave and IPFS have a long history of successful partnerships with other projects and vendors. Brave now ships with built-in wallets for cryptocurrencies, private windows that leverage Tor, and a highly integrated VPN option. IPFS, meanwhile, has collaborated with such names as Microsoft (developing a [decentralized identity stack](https://blog.ipfs.io/2020-06-11-identity-ipfs-ion/)), Netflix (experimenting [fetching Docker images via IPFS](https://blog.ipfs.io/2020-02-14-improved-bitswap-for-container-distribution/)), and NixOS (decentralizing [source code and build products](https://blog.ipfs.io/2020-09-08-nix-ipfs-milestone-1/)).
+
+![](119-brave-how-we-built-this/ipfs-early.png)
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:20]: Needs fixing (does not load from gateway)

File: content/post/119-how-we-built-this.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+*Early experiment showcasing IPFS URI resolution in Brave*
+
+This integration between IPFS and Brave is itself the product of a long-running experimental collaboration that began in [2017](https://github.com/brave/browser-laptop/issues/9556#issuecomment-352453877), back in the days when the Brave UI was still powered by Muon (a fork of Electron). In fact, this initiative got as far as a [proof of concept implementation](https://github.com/brave/browser-laptop/issues/9556#issuecomment-369757871) that provided for the resolution of the IPFS URIs in Brave’s address bar!
+
+![](119-brave-how-we-built-this/ipfs-streaming.jpg)
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:25]: Needs fixing

File: content/post/119-how-we-built-this.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Over the next two years, the teams behind Brave and IPFS would continue to come together to work toward full IPFS compatibility within the browser. [New plans](https://github.com/brave/brave-browser/issues/819#issuecomment-415792868) were made, and contributors from both teams and [the broader community](https://github.com/brave/brave-core-crx-packager/pull/21) began to chart the path toward this vision. In the interim, control over the browser source code allowed the teams to more tightly integrate the IPFS Companion extension into Brave: the Chrome sockets API (not normally exposed to Chrome Apps) made it possible to [embed a js-ipfs node with true TCP transport](https://github.com/brave/brave-browser/issues/819#issuecomment-456039555) within the extension, and Brave updated their settings menu to include a [one-click install](https://github.com/brave/brave-browser/issues/819#issuecomment-552444341) of the Companion.
+
+![](119-brave-how-we-built-this/ipfs-install.png)
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:32]: Needs fixing

File: content/post/119-how-we-built-this.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+*One-click install of the IPFS Companion in the Brave settings menu*
+
+Eventually, a confluence of factors, including the [deprecation of the Chrome sockets API](https://9to5google.com/2020/01/15/google-killing-chrome-apps/), generated [a push](https://github.com/brave/brave-browser/issues/10220) to get a full IPFS node running, fully managed, within Brave. After six months of work, we have finally realized this long-term goal!
+
+## Architecture
+A key goal of this integration was to make the use of IPFS as seamless as possible for users, while also respecting and preserving their control over the browser. When users enter an <code>ipfs://</code> or <code>ipns://</code> URI into their address bar for the first time, Brave will issue a prompt, asking whether or not they would prefer to resolve the link using a [public IPFS gateway](https://docs.ipfs.io/concepts/ipfs-gateway/) (by default, Brave uses [https://dweb.link](), though users can configure this) or via their own local IPFS node, managed by Brave. Initiating a Brave-managed local node is also possible through the interface of the IPFS Companion extension.
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:38]: Link did not render correctly, also does not need to be a link:
```suggestion
A key goal of this integration was to make the use of IPFS as seamless as possible for users, while also respecting and preserving their control over the browser. When users enter an <code>ipfs://</code> or <code>ipns://</code> URI into their address bar for the first time, Brave will issue a prompt, asking whether or not they would prefer to resolve the link using a [public IPFS gateway](https://docs.ipfs.io/concepts/ipfs-gateway/) (by default, Brave uses `https://dweb.link`, though users can configure this) or via their own local IPFS node, managed by Brave. Initiating a Brave-managed local node is also possible through the interface of the IPFS Companion extension.
```

File: content/post/119-how-we-built-this.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+---
+date: 2021-01-21
+url: /2021-01-21-how-we-built-this/
+title: "Origin Stories: IPFS in Brave"
+author: Mitch Wagner
+tags: browsers
+header_image: how-we-built-this.png
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:7]: Does not load, needs fixing.

**Review by lidel — CHANGES_REQUESTED** [review]
The historical journey across github issues if fantastic illustration of effort across time, space and community :)

Some suggestions + images to fix inline.

File: content/post/119-how-we-built-this.md (hunk 7 above)
Comment (jennwrites) [reply content/post/119-how-we-built-this.md:7]: ahhh rats

File: content/post/119-how-we-built-this.md (hunk 3 above)
Comment (jennwrites) [reply content/post/119-how-we-built-this.md:20]: Are you commenting on the screenshots?

**Comment by lidel** [comment]
I've rebased this PR on master (which already had header image) +  fixed images, now all load fine. 

nit: I think we may need to tweak text in the header image: 
- use "How we put IPFS in Brave"  as H1
- "Integration" looks out of place in H2

@jennwrites are you able to tweak Figma source at https://github.com/protocol/design-shop/issues/469#issuecomment-764667713 or do we need to ping someone to make those changes for us?

File: content/post/119-how-we-built-this.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+tags: brave, browsers
+header_image: 119-how-we-put-ipfs-in-brave.jpg
+snippet: This integration marks a major milestone for IPFS, and lays the groundwork for further experiments into improving the experience of interacting with the network via web browsers.
+---
+
+<p style="text-align: center;">
+  <img src="/header_images/119-how-we-put-ipfs-in-brave.jpg" alt="Integrating IPFS and Brave – How We Built This">
+</p>
+
+You may have heard the news: as of [release 1.19.86](https://github.com/brave/brave-browser/releases/tag/v1.19.86), the Brave browser has officially added [InterPlanetary File System (IPFS)](https://blog.ipfs.io/2021-01-19-ipfs-in-brave/) support! This work is the result of a multi-year effort to bring the two projects together. In this post, we discuss the process in bringing this collaboration to fruition, and take a look under the hood to see how we accomplished this integration!
```
Comment (lidel) [inline content/post/119-how-we-built-this.md:15]: I collected the news in a single comment, is this useful? 
```suggestion
You may have heard [the news](https://github.com/ipfs/in-web-browsers/issues/64#issuecomment-763016248): as of [release 1.19.86](https://github.com/brave/brave-browser/releases/tag/v1.19.86), the Brave browser has officially added [InterPlanetary File System (IPFS)](https://blog.ipfs.io/2021-01-19-ipfs-in-brave/) support! This work is the result of a multi-year effort to bring the two projects together. In this post, we discuss the process in bringing this collaboration to fruition, and take a look under the hood to see how we accomplished this integration!
```

Alternatively, we could make each word in sentence to link to a different publication.

**Comment by jennwrites** [comment]
Whew, a true team effort! Thanks for playing everyone, here is the live link: https://blog.ipfs.io/2021-01-21-how-we-put-ipfs-in-brave/

**Pull request merged**